MyLibrary/MyLibraryApp.API/Controllers/BooksController.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using MyLibraryApp.Application.Models.Books;
+using MyLibraryApp.Application.Services.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MyLibraryApp.API.Controllers
+{
+    [ApiController]
+    [Route("Books")]
+    public class BooksController :ControllerBase
+    {
+        private readonly IBookService bookService;
+        public BooksController(IBookService bookService)
+        {
+            this.bookService = bookService;
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GetAllBooksAsync()
+        {
+            var response = await bookService.ReadAllBooksAsync();
+
+            return Ok(response);
+        }
+
+        [HttpGet("{id}")]
+        public async Task<IActionResult> GetBookByIdAsync(int id)
+        {
+            var response = await bookService.GetBookByIdAsync(id);
+
+            return Ok(response);
+        }
+        
+        [HttpGet("Author/{author}")]
+        public async Task<IActionResult> GetBookByAuthorAsync(string author)
+        {
+            var response = await bookService.GetAllBooksByAuthor(author);
+
+            return Ok(response);
+        }
+
+        [HttpGet("Title/{title}")]
+        public async Task<IActionResult> GetBookByTitleAsync(string title)
+        {
+            var response = await bookService.GetAllBooksByTitle(title);
+
+            return Ok(response);
+        }
+
+        [HttpGet("Category/{categoryId}")]
+        public async Task<IActionResult> GetBookByCategoryIdAsync(int categoryId)
```

Comment:
Again on consistency, we have two different approaches:
- get book by author and title are having a string parameter (author name, book title)
- get book by category is having an int parameter (category id)
In some scenarios it might be perfectly fine to do it this way but not this time. I suggest all of them using the it as parameter (author id, book id (which is already defined as endpoint), category id).